### DISCUSSION TITLE
What would happen if the server do not broadcast commits?

### DISCUSSION THREAD

**Discussion by etclub** [stepwisehq/prosemirror-collab-commit#3] [Q&A] [answered]
I wonder if I can use this module in a situation that the server do not broadcast commits. Instead:
1. every client gets the latest version document from the server when it logs in
2. the server deals with the commits, and send back confirmation commits back to the client (without broadcasting to other clients)

In this case, Client A may send commits oftenly, while client B may lag behind a lot.
When Client B send commits, is there any risk that the commits cannot be auto merged/rebased?

**Comment by ProTip** [comment, marked as answer]
The behavior in this regard can be considered identical to that of ProseMirror's official collab plugin, and all of the discussion on [discuss.prosemirror.net](discuss.prosemirror.net) about offline editing should apply.

Generally speaking, the longer the histories diverge the higher the likelihood an update may get dropped and/or the document structure will diverge from user intention.

As an example if client B spends a lot of time editing inside a paragraph that client A deleted an hour ago, client B will not know this until it receives the commits made by client A. The system will "work" but the user will have lost their edits. When clients are kept up-to-date more frequently the risk is minimized and humans can take corrective action sooner if something undesired occurs in the document.

This is why Google Docs, in the case of a communication disruption, immediately freezes editing and asks the user if they'd like to take the document "offline".
